### PR TITLE
Support to specify space name when creating a new space

### PR DESCRIPTION
**Pull request by yixinglu**
1. Migrate starts/ends with expression tests
2. Split Index feature to edge/tag index features
3. Fix fetch tests comparison method
4. Cleanup marks and fix some nGQLs format
5. Support `Examples` in feature file level

### CODE REVIEW

**Review by jievince — DISMISSED** [review]

**Review by jievince — DISMISSED** [review]

File: tests/tck/features/schema/Schema.feature [hunk 1]
```diff
@@ -537,61 +549,54 @@ Feature: Insert string vid of vertex and edge
...
       CREATE EDGE e(name string DEFAULT "N/A", age int DEFAULT -1)
       """
-    Then the execution should be successful
     When executing query:
       """
       ALTER EDGE e ADD (description string DEFAULT "none")
       """
     Then the execution should be successful
-    Given wait 3 seconds
-    When executing query:
```
Comment (laura-ding) [inline tests/tck/features/schema/Schema.feature:552]: why delete the case?

File: tests/tck/features/insert/Insert.IntVid.feature [hunk 2]
```diff
@@ -1,445 +0,0 @@
...
-    # """
-    # INSERT VERTEX person(name, age) VALUES uuid("Tony"):("Tony", 18), uuid("Mack"):("Mack", 19);
-    # INSERT VERTEX person(name, age) VALUES uuid("Mack"):("Mack", 20);
-    # INSERT VERTEX person(name, age) VALUES uuid("Mack"):("Mack", 21)
-    # """
-    # Then the execution should be successful
-    # insert multi version edge with uuid
-    # When executing query:
-    # """
-    # INSERT EDGE schoolmate(likeness, nickname) VALUES uuid("Tony")->uuid("Mack")@1:(1, "");
```
Comment (laura-ding) [inline tests/tck/features/insert/Insert.IntVid.feature:432]: why delete it? the uuid

File: tests/tck/features/insert/Insert.feature [hunk 3]
```diff
@@ -396,29 +432,47 @@ Feature: Insert string vid of vertex and edge
...
       """
-      GO FROM "Tony" OVER schoolmate YIELD $$.person.name, $$.person.age, schoolmate.likeness
+      GO FROM <vid_Tony> OVER schoolmate YIELD $$.person.name, $$.person.age, schoolmate.likeness
       """
     Then the result should be, in any order:
       | $$.person.name | $$.person.age | schoolmate.likeness |
       | 'Mack'         | 21            | 3                   |
     Then drop the used space
 
+    Examples: Vertical
```
Comment (laura-ding) [inline tests/tck/features/insert/Insert.feature:458]: Do you want to use the same feature file for int and string id tests ?

Comment (yixinglu) [reply tests/tck/features/insert/Insert.feature:458]: yeah

File: tests/tck/features/schema/Schema.feature (hunk 1 above)
Comment (yixinglu) [reply tests/tck/features/schema/Schema.feature:552]: Have tested in above scenario

File: tests/tck/features/insert/Insert.IntVid.feature (hunk 2 above)
Comment (yixinglu) [reply tests/tck/features/insert/Insert.IntVid.feature:432]: My mistakes. I thought these were same feature files except vid type, not notice the uuid tests. Thanks. Have restored them.

**Review by laura-ding — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Review by laura-ding — APPROVED** [review]

**Pull request merged**